src/oisst/State/State.cc

```diff
@@ -5,105 +5,358 @@
-                    __FILE__, __LINE__);
+    // Ligang: just consider 1 var for now
+
+    // atlasFieldSet_->field(0).array() = 0.0  // does not work.
+    // field_data = 0.0; // Ligang: undefined operator "="!
+
+    auto field_data = make_view<float, 1>(atlasFieldSet_->field(0));
+    field_data.assign(0.0);
   }
 
 // ----------------------------------------------------------------------------
 
   void State::read(const eckit::Configuration & conf) {
-     util::abor1_cpp("State::read() needs to be implemented.",
-                     __FILE__, __LINE__);
+    int iread = 0, time = 0, lon = 0, lat = 0, bc = 0;
+    std::string sdate, filename, record;
+
+    auto field_data = make_view<float, 1>(atlasFieldSet_->field(0));
+    const int SIZE = geom_->atlasFunctionSpace()->size();
+
+    // Ligang: conf only has "statefile" part of the yml file.
+    // Here we do not actually use iread
+    if (conf.has("read_from_file")) {
+      iread = conf.getInt("read_from_file");
+    } else {
+      // you can output keys to check
+      std::vector<std::string> keys = conf.keys();
+    }
+
+    // get filename
+    if (!conf.get("filename", filename))
+      util::abor1_cpp("Get filename failed.", __FILE__, __LINE__);
+
+    // Ligang: open netCDF file
+    netCDF::NcFile file(filename.c_str(), netCDF::NcFile::read);
+    if (file.isNull())
+      util::abor1_cpp("Create netCDF file failed.", __FILE__, __LINE__);
+
+    // Ligang: get file dimensions, for checking
+    time = static_cast<int>(file.getDim("time").getSize());
+    lon  = static_cast<int>(file.getDim("lon").getSize());
+    lat  = static_cast<int>(file.getDim("lat").getSize());
+    if (time != 1 ||
+        lat != static_cast<int>(geom_->atlasFunctionSpace()->grid().ny()) ||
+        lon != static_cast<int>((((atlas::RegularLonLatGrid&)  // LC: no &?
+               (geom_->atlasFunctionSpace()->grid()))).nx()) ) {
+      util::abor1_cpp("lat!=ny or lon!=nx", __FILE__, __LINE__);
+    }
+
+    // get sst data
+    netCDF::NcVar sstVar;
+    sstVar = file.getVar("sst");
+    if (sstVar.isNull())
+      util::abor1_cpp("Get sst var failed.", __FILE__, __LINE__);
+
+    // Ligang: do we have to use for-loop?
+    // sstVar.getVar(field_data); // doesn't work this way
+    float sstData[lat][lon];
+    sstVar.getVar(sstData);
+
+    int idx = 0;
+    for (int i = 0; i < lat; i++)
+      for (int j = 0; j < lon; j++)
+        field_data(idx++) = sstData[i][j];
   }
 
 // ----------------------------------------------------------------------------
 
   void State::write(const eckit::Configuration & conf) const {
-    util::abor1_cpp("State::write() needs to be implemented.",
-                     __FILE__, __LINE__);
+    int lat, lon, time = 1;
+    std::string filename;
+
+    // get filename
+    if (!conf.get("filename", filename))
+      util::abor1_cpp("Get filename failed.", __FILE__, __LINE__);
+    else
+      std::cout << "State::write(), filename=" << filename << std::endl;
```

Comment:
`std::cout` should not be used. Either `oops::Log::info() ` or `oops::Log::debug() `... i'll add an issue and we can deal with this later

Reply:
ok, will delete std::cout after the current version has been merged.